### PR TITLE
Datetime serialisation fix

### PR DESCRIPTION
**Pull request by amn41**
**Proposed changes**:
- this fixes a bug trying to JSON serialise stories containing `ReminderScheduled` events. 

Would throw an exception like:
`TypeError: datetime.datetime(2018, 2, 3, 16, 32, 10, 827475) is not JSON serializable`

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa_core/events/__init__.py [hunk 1]
```diff
@@ -475,7 +475,7 @@ def __str__(self):
     def as_story_string(self):
         props = json.dumps({
             "action": self.action_name,
-            "date_time": self.trigger_date_time,
+            "date_time": self.trigger_date_time.isoformat(),
```
Comment (tmbo) [inline rasa_core/events/__init__.py]: Need to parse this in the `_from_story_string` function again.

**Comment by tmbo** [comment]
Can you also please add a test for that function?

File: tests/test_events.py [hunk 2]
```diff
@@ -68,3 +68,34 @@ def test_event_has_proper_implementation(one_event, another_event):
...
+
+    ActionExecuted("my_action"),
+
+    BotUttered("my_text", "my_data"),
+
+    ReminderScheduled("my_action", "now")
+])
+def test_serialisation(one_event):
+    story_string = one_event.as_story_string()
+    new_event = Event.from_story_string(story_string, one_event.as_dict(), None)
```
Comment (amn41) [inline tests/test_events.py]: @tmbo should this work for all events? (does for reminderscheduled) - do we need a real `domain` for any of these?

Comment (tmbo) [reply tests/test_events.py]: Yes, I think at least `ActionExecuted` needs it. best is to add `default_domain` as a parameter - will be filled automatically and use that one. The default_domain also has a slot `name` that you can set to a value. Best is to also use an action that is in there, so e.g. `utter_greet`.

File: rasa_core/events/__init__.py (hunk 1 above)
Comment (tmbo) [reply rasa_core/events/__init__.py]: ✅

File: tests/test_events.py (hunk 2 above)
Comment (amn41) [reply tests/test_events.py]: alright will update that - didn't work for `TopicSet` but that might also be related to domain

**Pull request merged**